custom_components/zwave_mqtt/climate.py

```diff
@@ -0,0 +1,461 @@
+                break
+
+        current_mode_value = self._mode().value.get(VALUE_SELECTED)
+        if current_mode_value in HVAC_MODE_MAPPINGS:
+            self._hvac_mode = current_mode_value
+            self._preset_mode = PRESET_NONE
+        else:
+            current_mode_label = self._hvac_value_label_mapping.get(current_mode_value)
+            if (
+                "heat" in current_mode_label.lower()
+                and HVAC_MODE_HEAT in self._hvac_list
+            ):
+                self._hvac_mode = ZW_HVAC_MODE_HEAT
+            elif (
+                "cool" in current_mode_label.lower()
+                and HVAC_MODE_COOL in self._hvac_list
+            ):
+                self._hvac_mode = ZW_HVAC_MODE_COOL
+            else:
+                self._hvac_mode = self._default_hvac_mode
+            self._preset_mode = current_mode_value
+
+    def _update_current_temp(self):
+        """Update current temperature."""
+        if not self.values.temperature:
+            return
+        self._current_temperature = self.values.temperature.value
+        device_unit = self.values.temperature.units
+        if device_unit is not None:
+            self._unit = device_unit
+
+    def _update_fan_mode(self):
+        """Update fan mode."""
+        if not self.values.fan_mode:
+            return
+        self._fan_value_label_mapping = {}
+        self._fan_label_value_mapping = {}
+        self._fan_modes = []
+        for entry in self.values.fan_mode.value[VALUE_LIST]:
+            self._fan_value_label_mapping[entry.get(VALUE_ID)] = entry.get(VALUE_LABEL)
+            self._fan_label_value_mapping[entry.get(VALUE_LABEL)] = entry.get(VALUE_ID)
+            self._fan_modes.append(entry.get(VALUE_ID))
+        self._current_fan_mode = self.values.fan_mode.value.get(VALUE_SELECTED)
+
+    def _update_target_temp(self):
+        """Update target temperature."""
+        current_setpoints = self._current_mode_setpoints()
+        self._target_temperature = None
+        self._target_temperature_range = (None, None)
+        if len(current_setpoints) == 1:
+            setpoint = current_setpoints[0]
+            if setpoint is not None:
+                self._target_temperature = round((float(setpoint.value)), 1)
+        elif len(current_setpoints) == 2:
+            (setpoint_low, setpoint_high) = current_setpoints
+            target_low, target_high = None, None
+            if setpoint_low is not None:
+                target_low = round((float(setpoint_low.value)), 1)
+            if setpoint_high is not None:
+                target_high = round((float(setpoint_high.value)), 1)
+            self._target_temperature_range = (target_low, target_high)
+
+    def _update_operating_state(self):
+        """Update operating state."""
+        if not self.values.operating_state:
+            return
+        mode = self.values.operating_state.value
+        self._hvac_action = HVAC_CURRENT_MAPPINGS.get(str(mode).lower(), mode)
```

Comment:
We should set `hvac_action` to `None` if the mapping is missing the mode. Ie remove the default in `dict.get`.